Upload projects_ngis-cat_assets_DEA_NIDEM to the catalog

ShiruiH opened the pull request:
Hi Kurt,

Thank you for reviewing our previous PR #486.  I have made revisions and updated them here based on all your comments. This version of the changes is neater and tidier, which should make it easier to review. Would you mind if I close #486 and we continue with this PR?

The dataset is available in the GEE Assets of our Google Cloud Project (GCP): https://code.earthengine.google.com/?asset=projects/ngis-cat/assets/DEA/NIDEM

### Key replies:

- We kept the name of 'ngis-cat' for now since it is the name of the GCP where we store the dataset. Will it be ok to change this subdirectory to another name? I found it reports bugs when using name differs from it.

- The fields `platform` and `instruments` have been removed for now, since the dataset is an inter tidal DEM product developed by Digital Earth Australia ([the product page](https://cmi.ga.gov.au/data-products/dea/325/dea-intertidal-elevation-landsat)). However, the data was developed based on Landsat . I'm not sure if the `instruments` field should be set to `'Landsat'` in this case.

schwehr commented on catalog/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.jsonnet:
```diff
@@ -0,0 +1,151 @@
+        'gee:units': units.meters,
+      },
+    ],
+    nidem: {minimum: -5.0, maximum: 3.9, 'gee:estimated_range': true},
```
If you know the actual min/max values, you can set estimated to `false`

schwehr replied:
Please project min/max for the uncertainty and unfiltered bands

ShiruiH commented on catalog/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.jsonnet:
```diff
@@ -0,0 +1,151 @@
+      {
+        name: 'nidem_mask',
+        description: |||
+          Mask - 25 m to 25 m relative to Mean Sea Level. 
```
Thanks for pointing this out. I checked the dataset and can see it flagged the above 25 m pixels as value 1, the below -25 m area as value 2, and the invalid elevation estimates as value 3. It didn't appear in its original descriptions provided by DEA. So I've added the `gee:classes` field to specify the values.

ShiruiH commented:
Hi Kurt,

Thanks for reviewing my PR.

I've addressed all your comments in the latest commit. Would you mind to check the changes and let me know if further actions needed?

### Main changes:
1. Directory names under catalog/ and example/ were changed to DEA/
2. Filenames under the above-mentioned directories were changed to DEA_NIDEM
3. Field `gee:classes` was added to describe the `nidem_mask` band

Cheers,
Shirui

ShiruiH reviewed (commented):
All requests resolved.

schwehr commented on catalog/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.jsonnet:
```diff
@@ -0,0 +1,166 @@
+    'dem',
+    'nidem',
+  ],
+  providers: [
```
Add NGIS as a processor. NGIS did the upload, wrote the STAC Collection jsonnet, and wrote the example.

```
  providers: [
    ee.producer_provider(
      'Digital Earth Australia',
      'https://cmi.ga.gov.au/data-products/dea/325/dea-intertidal-elevation-landsat'),
    ee.processor_provider('NGIS', 'https://ngis.com.au/'),
    ee.host_provider(self_ee_catalog_url),
  ],
```

schwehr commented on catalog/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.jsonnet:
```diff
@@ -0,0 +1,166 @@
+          Sea Level. 
+          [Band details](https://cmi.ga.gov.au/data-products/dea/325/dea-intertidal-elevation-landsat#details)
+        |||,
+        'gee:units': units.meters,
```
`meters` → `meter` here and below. You will need to rebase or merge from google:main for this to work

schwehr commented on catalog/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.jsonnet:
```diff
@@ -0,0 +1,166 @@
+    nidem: {minimum: -5.0, maximum: 3.9, 'gee:estimated_range': false},
+    nidem_uncertainty: {minimum: 0, maximum: 0.3, 'gee:estimated_range': false},
+    nidem_unfiltered: {minimum: -5.0, maximum: 3.9, 'gee:estimated_range': false},
+    'gee:visualizations': [
```
Tighten up the visualization section

```
    'gee:visualizations': [
      {
        display_name: 'nidem',
        lookat: {lon: 133.88, lat: -23.70, zoom: 5},
        image_visualization: {
          band_vis: {
            min: [-5.0],
            max: [3.9],
            palette: [
              '440154', '471365', '482475', '463480', '414487', '3b528b',
              '355f8d', '2f6c8e', '2a788e', '25848e', '21918c', '1e9c89',
              '22a884', '2fb47c', '44bf70', '5ec962', '7ad151', '9bd93c',
              'bddf26', 'dfe318', 'fde725',
            ],
            bands: ['nidem']}},
      },
    ],
  },
```

schwehr commented on catalog/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.jsonnet:
```diff
@@ -0,0 +1,166 @@
+    'gee:visualizations': [
+      {
+        display_name: 'nidem',
+        lookat: {lat: -23.70, lon: 133.88, zoom: 5},
```
Use lon, lat to match [Map.setCenter](https://developers.google.com/earth-engine/apidocs/map-setcenter)

schwehr commented on catalog/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.jsonnet:
```diff
@@ -0,0 +1,166 @@
+    Bishop-Taylor, R., Sagar, S., Lymburner, L., & Beaman, R. J. (2019).
+    Between the tides: Modelling the elevation of Australias exposed intertidal 
+    zone at continental scale. 
+    Estuarine, Coastal and Shelf Science, 223, 115 128. 
```
Fix the pages. It's missing the dash: `115-128`

schwehr commented on catalog/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.jsonnet:
```diff
@@ -0,0 +1,166 @@
+    ],
+  },
+  'sci:doi': '10.1016/j.ecss.2019.03.006',
+  'sci:citation': |||
```
Reflow the citation:

```
  'sci:citation': |||
    Bishop-Taylor, R., Sagar, S., Lymburner, L., & Beaman, R. J. (2019).
    Between the tides: Modelling the elevation of Australia's exposed intertidal
    zone at continental scale.  Estuarine, Coastal and Shelf Science, 223,
    115-128.
    [doi:10.1016/j.ecss.2019.03.006](https://doi.org/10.1016/j.ecss.2019.03.006)
  |||,
```

schwehr commented on checker/tree/keywords.py:
Revert this file

schwehr commented on catalog/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.jsonnet:
```diff
@@ -0,0 +1,166 @@
+    nidem_unfiltered: {minimum: -5.0, maximum: 3.9, 'gee:estimated_range': false},
+    'gee:visualizations': [
+      {
+        display_name: 'nidem',
```
Write out the dataset name and then give acronym.

```
        display_name: 'National Intertidal Digital Elevation Model (NIDEM)',
```

schwehr reviewed (changes requested):
I made https://github.com/google/earthengine-catalog/pull/548 for myself to work through the contents of this pr.  The summary of changes copied from there:

- Use only lf files (not dos \r\n) `dos2unix */ngis-cat/*`
- strip trailing space characters:`perl -pi -e 's/\s+\n/\n/g' */ngis-cat/*`
- Remove print from example
- `units.meters` → `units.meter`. I fixed that in https://github.com/google/earthengine-catalog/commit/7fc5af13543330a16e531ae05f92e7120986a01a
- add gee:skip_indexing and commented out the entry in ngis-cat/catalog.jsonnet
- comment out new keywords. keywords are meant to join multiple datasets
- reflow text
- Add NGIS as `processor_provider`
- Improve the visualization `display_name` and the example's `Map.addLayer`'s name
- Reflow the citation text, but keep doi link on a separate line
- Fixed citation pages to have a dash (`-`)
- Fixed citation text for `Australia's`
- Add links for `cite-as` and `source`

schwehr reviewed (changes requested):
I made https://github.com/google/earthengine-catalog/pull/548 for myself to work through the contents of this pr.  The summary of changes copied from there:

- Use only lf files (not dos \r\n) `dos2unix */ngis-cat/*`
- strip trailing space characters:`perl -pi -e 's/\s+\n/\n/g' */ngis-cat/*`
- Remove print from example
- `units.meters` → `units.meter`. I fixed that in https://github.com/google/earthengine-catalog/commit/7fc5af13543330a16e531ae05f92e7120986a01a
- add gee:skip_indexing and commented out the entry in ngis-cat/catalog.jsonnet
- comment out new keywords. keywords are meant to join multiple datasets
- reflow text
- Add NGIS as `processor_provider`
- Improve the visualization `display_name` and the example's `Map.addLayer`'s name
- Reflow the citation text, but keep doi link on a separate line
- Fixed citation pages to have a dash (`-`)
- Fixed citation text for `Australia's`
- Add links for `cite-as` and `source`

schwehr commented on catalog/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.jsonnet:
```diff
@@ -0,0 +1,165 @@
+    The National Intertidal Digital Elevation Model (NIDEM; Bishop-Taylor et
+    al. 2018, 2019) is a continental-scale elevation dataset for Australia's
+    exposed intertidal zone.
+    NIDEM provides the first three-dimensional representation of Australia's
```
Reflow.

```
     exposed intertidal zone.  NIDEM provides the first three-dimensional
     representation of Australia's intertidal sandy beaches and shores, tidal
     flats and rocky shores and reefs at 25 m spatial resolution, addressing a
     key gap between the availability of sub-tidal bathymetry and terrestrial
     elevation data. NIDEM was generated by combining global tidal modelling with
     a 30-year time series archive of spatially and spectrally calibrated Landsat
     satellite data managed within the Digital Earth Australia (DEA) platform.
```

schwehr commented on catalog/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.jsonnet:
```diff
@@ -0,0 +1,165 @@
+  |||,
+  license: license.id,
+  links: ee.standardLinks(subdir, id) + [
+      ee.link.license(license.reference),
```
Only indent by 2 spaces.

schwehr commented on catalog/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.jsonnet:
```diff
@@ -0,0 +1,165 @@
+        name: 'nidem',
+        description: |||
+          Primary product: elevation in meter units relative to modelled Mean
+          Sea Level.
```
`Sea Level` → `Mean Sea Level (MSL)`

Assuming this isn't using one of the other sea levels like MLLW, MLW, MHW, MHHW, ...

Looking below, it looks like this is indeed MSL.

ShiruiH replied:
The description shown here is "Mean Sea Level". I've added the acronym after it.

schwehr commented on catalog/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.jsonnet:
```diff
@@ -0,0 +1,165 @@
+    ],
+    nidem: {minimum: -5.0, maximum: 3.9, 'gee:estimated_range': false},
+    nidem_uncertainty: {minimum: 0, maximum: 0.3, 'gee:estimated_range': false},
+    nidem_unfiltered: {minimum: -5.0, maximum: 3.9, 'gee:estimated_range': false},
```
Fit to 80 columns:

```
    nidem_unfiltered: {
      minimum: -5.0, maximum: 3.9, 'gee:estimated_range': false},
```

schwehr commented on examples/ngis-cat/projects_ngis-cat_assets_DEA_NIDEM.js:
```diff
@@ -0,0 +1,14 @@
+  ],
+};
+Map.setCenter(122.36, -18.10, 11);
+Map.addLayer(elevation, elevationVis, 'nidem');
```
Give a more detailed layer name and include units:

```
Map.addLayer(
    elevation, elevationVis,
    'National Intertidal Digital Elevation Model (NIDEM; m)');
```
